### ISSUE TITLE
Debugger requires path to program be correct casing for breakpoints to get hit

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode-node-debug#26]
_From @martellaj on December 9, 2015 20:59_

Assume there's a file `app.js` in directory `Foo`. If in my `launch.json` file, I specify `foo/app.js` as the `program`, the debugger will run, but none of my breakpoints will get hit. However, if I give it the actual casing of the path, `Foo/app.js`, the debugger runs and breakpoints will get hit.

I think it should be all or nothing here. As in, if correct casing is required for breakpoints to get hit, then correct casing should be required for the debugger to work at all. Or vice versa (and preferably), if correct casing isn't required to run the debugger, than correct casing shouldn't be required for breakpoints to get hit.

_Copied from original issue: Microsoft/vscode#1149_

**Comment by martellaj** [comment]
@weinand, thanks for porting over. I see you closed the issue, but the resolution is unclear.

**Comment by weinand** [comment]
@martellaj my "Issue Mover" tool closed the issue in the vscode repo and moved it to the vscode-node-debug repo. But here the issue is still **open** and there is no resolution yet.

**Comment by martellaj** [comment]
Oh okay, I could've sworn it was closed here, too. Time to go get :coffee:. Thanks!

**Comment by weinand** [comment]
@martellaj on a case-sensitive filesystem (e.g. on linux) your setup "foo/app.js" will not work if the file on disk actually lives in a folder "Foo". So I would call your setup at least 'dangerous' (and I was hit by this many times because I develop on a case insensitive file system but run builds on linux).

So from this perspective, I would rather try to flag the case-mismatch of the 'program' attribute as an error than make breakpoints work in this case. In addition another problem lurks around the corner: if you `require('App.js')`this will work on a case insensitive filesystem but again the breakpoints in 'app.js' will not be hit, because for node and the debugger the source lives in 'App.js'.

Bottom line: I will not make breakpoint matching case insensitive because it will hide potential problems. Instead I will make the initial case mismatch in your case an error because it will prevent subsequent breakpoints from being hit.

@martellaj what do you think?

**Comment by martellaj** [comment]
@weinand I think that's a good solution. Like I said initially, all or nothing is definitely better IMO. :+1:
